astropy/wcs/wcsapi/low_level_api.py

```diff
@@ -178,6 +181,12 @@ def world_axis_object_classes(self):
         * The last tuple element must be a dictionary with the keyword
           arguments required to initialize the class.
 
+        For advanced use cases, the tuple can also contain four elements, where
```

Comment:
I like this, to make it an extra, because then the type information is preserved. Why don't we add it at the end though, otherwise you end up with having to do backwards indexing to make things reliable?

Reply:
Yes that could work and simplify things

Reply:
Done